src/simple_api.rs

```diff
@@ -0,0 +1,267 @@
+    pub initial_window_height: f32,
+    /// if you want to adjust your imgui window to match the size of the outer window
+    /// this makes it possible to have a "fullscreen" imgui window spanning the whole current window.
+    pub on_resize: &'static dyn Fn(&winit::dpi::PhysicalSize<u32>, &mut State, f64),
+    /// called after the premade events have been handled which includes close request
+    /// if you think you need to handle this, this api abstraction is probably to high level
+    /// and you may want to copy the code from hello_world.rs and adapt directly
+    pub on_event: &'static dyn Fn(&winit::event::WindowEvent<'_>, &mut State),
+    /// font size
+    pub font_size: Option<f32>,
+    /// color that fills the window
+    pub background_color: wgpu::Color,
+}
+
+impl<State> Default for Config<State> {
+    fn default() -> Self {
+        Self {
+            window_title: "imgui".to_string(),
+            initial_window_width: 1200.0,
+            initial_window_height: 720.0,
+            on_resize: &|_, _, _| {},
+            on_event: &|_, _| {},
+            font_size: None,
+            background_color: wgpu::Color {
+                r: 0.0,
+                g: 0.0,
+                b: 0.0,
+                a: 1.0,
+            },
+        }
+    }
+}
+
+/// simple function to draw imgui
+pub fn run<YourState: 'static, UiFunction: 'static + Fn(&imgui::Ui, &mut YourState)>(
+    config: Config<YourState>,
+    mut state: YourState,
+    render_ui: UiFunction,
+) {
+    // Set up window and GPU
+    let event_loop = EventLoop::new();
+
+    let instance = wgpu::Instance::new(wgpu::BackendBit::PRIMARY);
+
+    let (window, size, surface) = {
+        let window = Window::new(&event_loop).unwrap();
+        window.set_inner_size(LogicalSize {
+            width: config.initial_window_width,
+            height: config.initial_window_height,
+        });
+        window.set_title(&format!("{}", config.window_title));
+        let size = window.inner_size();
+
+        let surface = unsafe { instance.create_surface(&window) };
+
+        (window, size, surface)
+    };
+
+    let hidpi_factor = window.scale_factor();
+
+    let adapter = block_on(instance.request_adapter(&wgpu::RequestAdapterOptions {
+        power_preference: wgpu::PowerPreference::HighPerformance,
+        compatible_surface: Some(&surface),
+    }))
+    .unwrap();
+
+    let (device, queue) = block_on(adapter.request_device(
+        &wgpu::DeviceDescriptor {
+            features: wgpu::Features::empty(),
+            limits: wgpu::Limits::default(),
+            shader_validation: false,
+        },
+        None,
+    ))
+    .unwrap();
+
+    // Set up swap chain
+    let sc_desc = wgpu::SwapChainDescriptor {
+        usage: wgpu::TextureUsage::OUTPUT_ATTACHMENT,
+        format: wgpu::TextureFormat::Bgra8UnormSrgb,
+        width: size.width as u32,
+        height: size.height as u32,
+        present_mode: wgpu::PresentMode::Mailbox,
```

Comment:
Prefer FIFO, limits refresh rate to the monitor's refresh rate, not wasting power spinning very quickly.

Reply:
Well, I personnally prefer Mailbox to see how it performs, but when it's not changeable I think FIFO might indeed be better :)

Reply:
Yeah the better option is FIFO by default, with an option to enable mailbox.

Reply:
should this be changed for hello world as well? probably not a big deal so I'll just leave it for now?